### PR TITLE
Off heap combined pr

### PR DESCRIPTION
**Pull request by rmnattas**

### CODE REVIEW

File: compiler/optimizer/Inliner.cpp [hunk 1]
```diff
@@ -3325,7 +3325,7 @@ TR::TreeTop * OMR_InlinerUtil::storeValueInATemp(
       bool isInternalPointer = false;
 
       // disable internal pointer symbols for C/C++
-      if ((value->hasPinningArrayPointer() &&
+      if (( (value->hasPinningArrayPointer()) &&
```
Comment (VermaSh) [inline compiler/optimizer/Inliner.cpp]: Can you please remove the extra brace

File: compiler/optimizer/InductionVariable.cpp [hunk 2]
```diff
@@ -3394,35 +3396,42 @@ bool TR_LoopStrider::reassociateAndHoistComputations(TR::Block *loopInvariantBlo
...
+         pinningArrayNode = node->getFirstChild()->getFirstChild();
       else
-         {
-         SymRefPair *pair = _parmAutoPairs;
-         while (pair)
+         pinningArrayNode = node->getFirstChild();
+
+      if (pinningArrayNode->getOpCode().isLoadVar() &&
+         pinningArrayNode->getSymbolReference()->getSymbol()->isAutoOrParm() &&
+         //pinningArrayNode->getSymbolReference()->getSymbol()->isAuto() &&
```
Comment (VermaSh) [inline compiler/optimizer/InductionVariable.cpp]: Do we need this commented line?

File: compiler/optimizer/InductionVariable.cpp [hunk 3]
```diff
@@ -3643,6 +3652,78 @@ bool TR_LoopStrider::reassociateAndHoistComputations(TR::Block *loopInvariantBlo
...
+            else
+               symbol->castToInternalPointerAutoSymbol()->setPinningArrayPointer(pinningArrayPointer->castToInternalPointerAutoSymbol()->getPinningArrayPointer());
+
+            (*_reassociatedAutos)[originalInternalPointerSymbol] = newSymbolReference;
+            dumpOptDetails(comp(), "reass num %d newsymref %d\n", internalPointerSymbol, newSymbolReference->getReferenceNumber());
+            }
+
+         TR::SymbolReference *internalPointerSymRef = (*_reassociatedAutos)[originalInternalPointerSymbol];
+         originalNode->getFirstChild()->recursivelyDecReferenceCount();
+         //node->recursivelyDecReferenceCount();
```
Comment (VermaSh) [inline compiler/optimizer/InductionVariable.cpp]: Do we need to keep this line around?

File: compiler/optimizer/LoopVersioner.cpp [hunk 4]
```diff
@@ -8431,115 +8434,131 @@ bool TR_LoopVersioner::depsForLoopEntryPrep(
...
                {
-               TR::Node *constNode = TR::Node::create(node, type.isInt32() ? TR::iconst : TR::lconst, 0, 0);
-               indexNode = TR::Node::create(type.isInt32() ? TR::isub : TR::lsub, 2, offset, constNode);
-               if (type.isInt64())
-                  constNode->setLongInt((int64_t)headerSize);
+               /* No need to subtract header size if using dataAddr pointer node. Since
+               * we already have a pointer to first array element the header size was never
+               * added to the offset.
+               * For Reference see Walker.cpp:calculateElementAddressInContiguousArray.
+               */
```
Comment (VermaSh) [inline compiler/optimizer/LoopVersioner.cpp]: Can you please fix the formatting here.

File: compiler/optimizer/LoopVersioner.cpp [hunk 5]
```diff
@@ -8431,115 +8434,131 @@ bool TR_LoopVersioner::depsForLoopEntryPrep(
...
-         }
 
 
-     // sverma: the index node didn't match either of the above (ishl/imul) patterns
-     //        so let's reverse the operations on offset to get the index value
-     if (!indexNode)
-         {
-         if (!childInRequiredForm)
+         // sverma: the index node didn't match either of the above (ishl/imul) patterns
+         //        so let's reverse the operations on offset to get the index value
```
Comment (VermaSh) [inline compiler/optimizer/LoopVersioner.cpp]: Do we need to keep this around?

**Review by VermaSh — CHANGES_REQUESTED** [review]
Looks good for the most part, just a few minor things.

**Comment by rmnattas** [comment]
Fixed them 👍

**Pull request merged**